lib/active_model/serializer/array_serializer.rb

```diff
@@ -5,7 +5,7 @@ class ArraySerializer
       include Enumerable
       delegate :each, to: :@objects
 
-      attr_reader :root, :meta, :meta_key
+      attr_reader :root, :meta, :meta_key, :options, :resource
```

Comment:
I totally agree when we are talking about `pagination` key, but there are also `original_url`and `query_parameters`. Do you think it should go in adapters hash as well?(i.e https://github.com/rails-api/active_model_serializers/blob/03ddeeeec1e4eedafeb68eeccd59e8c3826b859d/lib/active_model/serializable_resource.rb#L5), 

Thus there is no way to `JsonApi` adapter  know about the resources (collection). We can use method `each` which means exactly the same due to this delegate https://github.com/rails-api/active_model_serializers/blob/03ddeeeec1e4eedafeb68eeccd59e8c3826b859d/lib/active_model/serializer/array_serializer.rb#L6. However, I think it quite weird something like it 
```ruby
JsonApi::PaginationLinks.new(serializer.each, options)
```